iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/ConnectionShuntModification.java

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.iidm.modification;
+
+import com.powsybl.commons.reporter.Reporter;
+import com.powsybl.computation.ComputationManager;
+import com.powsybl.iidm.network.Network;
+import com.powsybl.iidm.network.Terminal;
+
+import java.util.Objects;
+
+/**
+ * Network modification to connect or disconnect a Shunt compensator.
+ *
+ * @author Nicolas Pierre <nicolas.pierre at artelys.com>
+ */
+public class ConnectionShuntModification extends AbstractNetworkModification {
```

Comment:
I think that you have to use the existing `ShuntCompensatorPositionModification` because the connection of the shunt could be useful. But most of the time, you want to reconnect a shunt at a dedicated section and not to the previous one.